proposals/safe-fixed-size-buffers.md

```diff
@@ -0,0 +1,378 @@
+    ;    
+```
+
+A *fixed_size_buffer_declarator* introduces a fixed-size buffer of a given element type.
+
+The buffer element type is the *type* specified in `field_declaration`. A fixed-size buffer declarator introduces a new member and consists of an identifier that names the member, followed by a constant expression enclosed in `[` and `]` tokens. The constant expression denotes the number of elements in the member introduced by that fixed-size buffer declarator. The type of the constant expression must be implicitly convertible to type `int`, and the value must be a non-zero positive integer.
+
+The elements of a fixed-size buffer shall be laid out sequentially in memory as though they are elements of an array.
+
+A *field_declaration* with a *fixed_size_buffer_declarator* in an interface must have `static` modifier.
+
+Depending on the situation (details are specified below), an access to a fixed-size buffer member is classified as a value (never a variable) of either
+`System.ReadOnlySpan<S>` or `System.Span<S>`, where S is the element type of the fixed-size buffer. Both types provide indexers returning a reference to a
+specific element with appropriate "readonly-ness", which prevents direct assignment to the elements when language rules don't permit that.
+
+This limits the set of types that can be used as a fixed-size buffer element type to types that can be used as type arguments. For example, a pointer type cannot be used as an element type.
+
+The resulting span instance will have a length equal to the size declared on the fixed-size buffer.
+Indexing into the span with a constant expression outside of the declared fixed-size buffer bounds is a compile time error.
+
+The *safe-to-escape* scope of the value will be equal to the *safe-to-escape* scope of the container, just as it would if the backing data was accessed as a field.
+
+### Fixed-size buffers in expressions
+
+Member lookup of a fixed-size buffer member proceeds exactly like member lookup of a field.
+
+A fixed-size buffer can be referenced in an expression using a *simple_name* or a *member_access* .
+
+When an instance fixed-size buffer member is referenced as a simple name, the effect is the same as a member access of the form `this.I`, where `I` is the fixed-size buffer member. When a static fixed-size buffer member is referenced as a simple name, the effect is the same as a member access of the form `E.I`, where `I` is the fixed-size buffer member and `E` is the declaring type.
+
+#### Non-readonly fixed-size buffers
+
+In a member access of the form `E.I`, if `E` is of a struct type and a member lookup of `I` in that struct type identifies a non-readonly instance fixed-size member,
+then `E.I` is evaluated and classified as follows:
+
+- If `E` is classified as a value, the result of the expression is classified as a value of type `System.ReadOnlySpan<S>`, where S is the element type of `I`.
+  The value can be used to access members’ elements.
```

Comment:
The value can be used to access memebr's elements but not must be. For example my reading believes the following should work: 

```csharp
S local = default;
Span<int> span = s.Field;

struct S {
  public int[5] Field;
}

```

Is that correct?

Reply:
> The value can be used to access memebr's elements but not must be. For example my reading believes the following should work:

Yes, ```Span<int> span = s.Field``` will work.